### PR TITLE
Refs #24928 -- Added introspection support for PostgreSQL RangeField

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/inspectdb/tests.py [hunk 1]
```diff
@@ -48,7 +48,10 @@ def assertFieldType(name, definition):
         return assertFieldType
 
     def test_field_types(self):
-        """Test introspection of various Django field types"""
+        """
+        Test introspection of various Django field types.
+        See also postgres_tests.test_introspection for PostgreSQL-specific fields.
```
Comment (timgraham) [inline tests/inspectdb/tests.py]: I don't think this change is needed. For example, there isn't a similar comment in the model field and form field tests.

File: tests/postgres_tests/test_introspection.py [hunk 2]
```diff
@@ -24,3 +24,15 @@ def test_json_field(self):
             'postgres_tests_jsonmodel',
             ['field = django.contrib.postgresql.fields.JSONField(blank=True, null=True)'],
         )
+
+    def test_range_introspection(self):
```
Comment (timgraham) [inline tests/postgres_tests/test_introspection.py]: test_range_fields

File: django/contrib/postgres/apps.py [hunk 3]
```diff
@@ -18,6 +18,11 @@ def ready(self):
             if conn.vendor == 'postgresql':
                 conn.introspection.data_types_reverse.update({
                     3802: 'django.contrib.postgresql.fields.JSONField',
+                    3904: 'django.contrib.postgresql.fields.IntegerRangeField',
+                    3906: 'django.contrib.postgresql.fields.FloatRangeField',
+                    3910: 'django.contrib.postgresql.fields.DateTimeRangeField',
+                    3912: 'django.contrib.postgresql.fields.DateRangeField',
+                    3926: 'django.contrib.postgresql.fields.BigIntegerRangeField',
                 })
```
Comment (adamchainz) [inline django/contrib/postgres/apps.py]: I've realized something about this - if, for whatever reason (perhaps custom subclassing??), multiple apps are installed that can introspect these types, if this pattern is copied it's the *last* app in `INSTALLED_APPS` that take precedence, even though the general rule with Django is apps earlier in `INSTALLED_APPS` take precedence. Idk if it's a huge issue though.

Comment (claudep) [reply django/contrib/postgres/apps.py]: Don't talk to that awkward first-winning ordering to me :-)

Comment (adamchainz) [reply django/contrib/postgres/apps.py]: 🤐

**Pull request merged**